### ISSUE TITLE
predictions after scvi integration

### ISSUE THREAD

**Issue by pirlowang** [Teichlab/celltypist#145]
I have run scvi integration for my 1.5 millions scRNA-seq data from multiple patients. when I run predictions = celltypist.annotate(adata, model = 'Adult_Human_Skin.pkl', majority_voting = True). It warning:
WARNING:celltypist.logger:⚠️ Warning: invalid expression matrix, expect all genes and log1p normalized expression to 10000 counts per cell. The prediction result may not be accurate

Then I run   adata = adata.raw.to_adata() and rerun annotation. It goes well but running leiden clustering again. I never get this run done because the kernal is killed even if I set 100G memory.  

The scvi command:
adata.layers['raw_counts']=adata.X.copy()
sc.pp.normalize_total(adata,target_sum=1e4)
sc.pp.log1p(adata)
adata.raw = adata
adata.raw = adata
scvi.model.SCVI.setup_anndata(adata, layer="raw_counts", batch_key='donor', categorical_covariate_keys=['platform'])
model = scvi.model.SCVI(adata, n_layers=1, n_latent=6, gene_likelihood="zinb")
model.train()
sc.pp.neighbors(adata, use_rep=SCVI_LATENT_KEY)
sc.tl.leiden(adata)
sc.tl.umap(adata)

**Comment by ChuanXu1** [comment]
@pirlowang, what's the shape of your raw count matrix? Does it contain only HVGs or all genes?

**Comment by pirlowang** [comment]
adata has a layer 'raw_counts', and adata.raw.X contains all normalized gene counts. adata.X has only HVGs. here is my HVGs command: 
sc.pp.highly_variable_genes(
    adata,
    flavor="seurat_v3",
    n_top_genes=2000,
    layer="counts",
    batch_key="batch",
    subset=True,
)

**Comment by ChuanXu1** [comment]
@pirlowang, you need to predict cell types based on all genes in your adata to ensure a maximum overlap with genes in the model. That's why after `subset = True`, CellTypist will output a warning as you have seen. The error you encountered is related to memory issue caused by the majority voting process which runs high-resolution Leiden again. You can either reduce the size of your data (e.g., after adata = adata.raw.to_adata(), remove the raw count by del adata.layers['raw_counts'], wrtie out the adata, and open a new session to load this smaller data). Otherwise, you have to turn off majority voting by `majority_voting = False` or rely on your own clustering by `majority_voting = your_adata_leiden_column`
